### PR TITLE
Implement real time updates

### PR DESCRIPTION
**Pull request by cp-sneha-s**
### Purpose
 - add real-time updates for admin screens(home,leave,members)

### Summary of Changes
 - Implement real-time update for admin home
 - implement real-time update for admin leave
 - implement rea-time update for admin members
 - Refactor unit test.
 - Add leave and members real-time update repo

### Test steps
 - Run flutter test

### Conformity

- [x] Followed [git guidelines](https://github.com/canopas/flutter-developer-roadmap/blob/main/GitGuideline.md) for creating commit messages and Pull Request guidelines.
- [x] Self-approved the PR - reviewed the PR as a reviewer and gave it self-approval if everything is ok. If not, made the required changes.
- [x] Ensured that the PR satisfies all specified requirements in the ticket, including bug fixes and new features.
- [x] Provided test steps, including steps to reproduce the issue or test the new functionality, ensuring other team members can verify the changes.
- [x] Added/Updated proper code comments to make it easy-to-understand for other developers.
- [x] Reused code (if the same code was written twice, made it common and reused it at both places).
- [x] Removed unused or commented code if not required.
- [x] Ensured proper Dart naming conventions were used for variables, classes, and methods.
- [x] Localized user-facing strings.
- [x] Included screenshots/videos of behavior changes: Provided visual evidence of any changes to UI or behavior for easier review and understanding in the PR description.
- [x] Implemented proper error handling: Ensured that the code anticipated and handled potential errors and edge cases gracefully.
- [x] Avoided introducing technical debt: If the PR introduces technical debt, created and linked appropriate tickets for future resolution.
- [x] Included relevant unit tests: Wrote unit tests that focused on testing behavior and functionality, rather than merely covering lines of code.
- [x] Ensured code was performant and scalable: Verified that the changes did not introduce performance issues or bottlenecks and could scale as needed.
- [x] Ensured comments were up-to-date and relevant to the code to describe complex logic and to add understanding for other developers.
- [x] Marked the PR as ready before submitting it for review.

### CODE REVIEW

File: lib/data/bloc/user_state/user_state_controller_bloc.dart [hunk 1]
```diff
@@ -27,16 +27,19 @@ class UserStateControllerBloc
       CheckUserStatus event, Emitter<UserControllerState> emit) async {
     try {
       final Employee? employee =
-          await _employeeService.getEmployee(_userManager.userUID!);
+          await _employeeService.getEmployee(_userStateNotifier.userUID!);
       final Space? space =
-          await _spaceService.getSpace(_userManager.currentSpaceId!);
+          await _spaceService.getSpace(_userStateNotifier.currentSpaceId!);
```
Comment (cp-sneha-s) [inline lib/data/bloc/user_state/user_state_controller_bloc.dart:32]: I think we have to add realtime listener

Comment (cp-pratik-k) [reply lib/data/bloc/user_state/user_state_controller_bloc.dart:32]: for what ??

Comment (cp-sneha-s) [reply lib/data/bloc/user_state/user_state_controller_bloc.dart:32]: For employee data and space data so, we don't have to listen to it repeatedly on the home screen.

Comment (cp-sneha-s) [reply lib/data/bloc/user_state/user_state_controller_bloc.dart:32]: Are you working on this?

**Review by cp-pratik-k — APPROVED** [review]
